### ISSUE TITLE
Adding node to AnimationTreePlayer and removing AnimationTreePlayer cause crash

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#46012]
**Godot version:**
Godot 3.2.4 rc 2

**Issue description:**
Executing
```
	var aa =  AnimationTreePlayer.new()
	aa.add_node(10,"")
	aa.queue_free()
```

crashes with backtrace
```
core/os/memory.h:117:14: runtime error: member call on null pointer of type 'struct NodeBase'
core/os/memory.h:117:14: runtime error: member access within null pointer of type 'struct NodeBase'
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] godots() [0x16f757a] (/mnt/Miecz/godot3.2/platform/x11/crash_handler_x11.cpp:54)
[2] /lib/x86_64-linux-gnu/libc.so.6(+0x46210) [0x7fdaf3116210] (??:0)
[3] void memdelete<AnimationTreePlayer::NodeBase>(AnimationTreePlayer::NodeBase*) (/mnt/Miecz/godot3.2/./core/os/memory.h:117)
[4] AnimationTreePlayer::~AnimationTreePlayer() (/mnt/Miecz/godot3.2/scene/animation/animation_tree_player.cpp:1864)
[5] void memdelete<Object>(Object*) (/mnt/Miecz/godot3.2/./core/os/memory.h:119)
[6] SceneTree::_flush_delete_queue() (/mnt/Miecz/godot3.2/scene/main/scene_tree.cpp:1115)
[7] SceneTree::iteration(float) (/mnt/Miecz/godot3.2/scene/main/scene_tree.cpp:492 (discriminator 5))
[8] Main::iteration() (/mnt/Miecz/godot3.2/main/main.cpp:2085)
[9] OS_X11::run() (/mnt/Miecz/godot3.2/platform/x11/os_x11.cpp:3634)
[10] godots(main+0x331) [0x16ee467] (/mnt/Miecz/godot3.2/platform/x11/godot_x11.cpp:57)
[11] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7fdaf30f70b3] (??:0)
[12] godots(_start+0x2e) [0x16ee07e] (??:?)
```

**Comment by angad-k** [comment]
May I work on this? Also, if yes then which are the files that I should look into to investigate the issue?

**Comment by angad-k** [comment]
I have identified and fixed the issue. Can I make a PR for the same?

**Comment by akien-mga** [comment]
Yes, go ahead :) You don't need to ask for permission, simply stating that you're working on the issue is enough (and if you stop working on it for one reason or another, you can update your comment to say so).

**Comment by akien-mga** [comment]
Fixed by #46113.
